config/groovy/common.groovy

```diff
@@ -188,6 +188,24 @@ class common {
      */
     def updateItem(String itemName) {
         File targetDir = new File(targetDirectory, itemName)
+        def searchString = itemName
+        if (itemName.startsWith(".")){
+            // add literal slash for regex start with '.'
+            searchString = "\\$itemName"        
+        }
+        def inGitIgnore = false
+        new File(".gitignore").eachLine{ line -> 
+            // match if line is exactly same os itemName or has trailing '/' but
+            // not if has further subdirectories
+            if ((line ==~ searchString) || (line ==~ "$searchString/")){
+                inGitIgnore = true
+                return
+            }
+        }
```

Comment:
This should be extracted to its own method for better readability. 

Doing so will also make it easier to spot bugs or find the right location for doing improvements. We're missing that `.gitignore` files can also appear in subfolders and "add up" to each other. Not a huge deal, but ideally we could use some library to do this work for us...

Reply:
Ok this now checks every `.gitignore`. I also added some output to tell you which `.gitignore` is blocking the module. See sample below:
![update-all-output-all-gitignores](https://user-images.githubusercontent.com/35314989/83468884-1da45480-a44c-11ea-9df2-a536d0121017.png)

Reply:
If I understand this correclty, `modules/foo` will be ignore now if mentioned in `sandbox/.gitignore`? I don't think that's how the ignored files are computed 🧐 

Some references:

- https://www.atlassian.com/git/tutorials/saving-changes/gitignore
- https://git-scm.com/docs/gitignore
- https://help.github.com/en/github/using-git/ignoring-files

Reply:
I'm not used to multiple .gitignores. I had a section computing the relative path but realized I made a mistake and was finding the relative path to the root dir instead of the modules dir. Then the build failed after backing out the relative path stuff which surprised me. Wanted to get more info on that before continuing the relative path